resources/js/mosaic.tile.js

```diff
@@ -1026,11 +1026,36 @@ class Tile {
             self.mosaic.saving_tile = true;
             utils.loading.show();
 
+            // Normalize internal resolveuid links before saving.
+            // In some editing contexts the editor can produce links that include
+            // the portal/site name (e.g. "/Plone/resolveuid/<uid>") or even the
+            // full absolute portal URL. Storing those absolute URLs can break the
+            // resolveuid transform when the viewing site URL differs from the
+            // editing URL. Convert such occurrences to a relative "../resolveuid/...".
+            let normalizedData = currentData;
+            try {
+                const portal = (self.mosaic && self.mosaic.options && self.mosaic.options.context_url) ? self.mosaic.options.context_url : null;
+                if (portal) {
+                    // escape portal for regex
+                    const escPortal = portal.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+                    // replace occurrences like "<portal>/resolveuid/..." with "../resolveuid/..."
+                    normalizedData = normalizedData.replace(new RegExp(escPortal + "\\/resolveuid\\/", "g"), "../resolveuid/");
+                    // also replace full urls prefixed with http(s)://host/.../resolveuid/
+                    normalizedData = normalizedData.replace(/https?:\/\/[^\s'"<>]+\/resolveuid\//g, "../resolveuid/");
+                }
+                // generic fallback: "/anySiteName/resolveuid/..." -> "../resolveuid/..."
+                normalizedData = normalizedData.replace(/(["'])\/[^\/\s'"<>]+\/resolveuid\//g, '$1../resolveuid/');
+            } catch (e) {
+                // if normalization fails for any reason, keep original HTML
+                log.warn(`Could not normalize resolveuid links: ${e}`);
+                normalizedData = currentData;
+            }
```

Comment:
I'm chewing on the 3 regexes and ask myself, if it can't just be combined into one?
In any case, we want the resolveuid to be "../resolveuid/".
Except maybe if the url directly starts with resolveuid, by maybe also those. Anyway, here it's left out.
But it would also catch "../resolveuid/" which would not be necessary, but I think doesn't hurt either.
So maybe:
```suggestion
            let normalizedData = currentData.replace(/(["'])[^'"]+\/resolveuid\//g, '$1../resolveuid/');
```

@petschki any opinions on that?